armi/reactor/grids/locations.py

```diff
@@ -0,0 +1,480 @@
+# Copyright 2023 TerraPower, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, TYPE_CHECKING, Union, Hashable, Tuple, List, Iterator
+from abc import ABC, abstractmethod
+import math
+
+import numpy
+
+if TYPE_CHECKING:
+    # Avoid some circular imports
+    from . import Grid
```

Comment:
I see you.

This is one of the options we've discussed. As I say, at some point we'll have to make the decision and we will unify all documentation in ARMI. 

But there are some discussions ahead first.

Reply:
And so it begins! 😈 😅